src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs

```diff
@@ -93,20 +92,18 @@ public IActionResult GetAddressIndexerTip()
         /// <returns><see cref="BlockModel"/> if block is found, <see cref="NotFoundObjectResult"/> if not found. Returns <see cref="IActionResult"/> with error information if exception thrown.</returns>
         [Route(BlockStoreRouteEndPoint.GetBlock)]
         [HttpGet]
-        public async Task<IActionResult> GetBlockAsync([FromQuery] SearchByHashRequest query)
+        public IActionResult GetBlock([FromQuery] SearchByHashRequest query)
         {
             if (!this.ModelState.IsValid)
-            {
                 return ModelStateErrors.BuildErrorResponse(this.ModelState);
-            }
 
             try
             {
                 Block block = this.blockStore.GetBlock(uint256.Parse(query.Hash));
 
                 if (block == null)
                 {
-                    return new NotFoundObjectResult("Block not found");
+                    return Ok("Block not found");
```

Comment:
why this change?

Reply:
`NotFoundObjectResult` creates 404 status codes, which relates to whether or not the caller could call the API end point. In this instance we did in fact call the endpoint successfully but didn't find the block, so the HTTP call is still valid.

Reply:
but AFAIK this is the standard behavior of REST APIs, so if we want to adhere to that standard, 404 is used to communicate that a resource is not found.
Basically in this case, if you ask for a block and it can't be find, it's not correct to return a successful message (status 200) 
e.g. take a look at this response https://stackoverflow.com/a/9930716/1716620

Consider that returning 404 is fine and we can add a content to that request that would be helpful to understand if the 404 was because of a failure on endpoint or because of a "block not found"

Reply:
as use case, even twitter uses 404
https://developer.twitter.com/en/docs/basics/response-codes.html